src/AdminBundle/Controller/PaperVoteController.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace AdminBundle\Controller;
+
+use AppBundle\Entity\Project;
+use AppBundle\Entity\VoteSettings;
+use AppBundle\Exception\ValidatorException;
+use Symfony\Bridge\Doctrine\Form\Type\EntityType;
+use Symfony\Component\Form\Extension\Core\Type\CollectionType;
+use Symfony\Component\Form\Extension\Core\Type\SubmitType;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
+use AppBundle\Entity\User;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter;
+
+/**
+ * User controller.
+ *
+ * @Route("/admin/vote")
+ */
+class PaperVoteController extends Controller
+{
+    /**
+     * @Route("/user/{user_id}/voting/{voting_id}/new", name="admin_user_new_paper_vote")
+     * @ParamConverter("user", class="AppBundle:User", options={"id" = "user_id"})
+     * @ParamConverter("voteSettings", class="AppBundle:VoteSettings", options={"id" = "voting_id"})
+     * @Method({"GET", "POST"})
+     * @Template()
+     *
+     * @param User $user
+     * @param VoteSettings $voteSettings
+     * @param Request $request
+     *
+     * @return array | bool | RedirectResponse
+     */
+    public function newAction(User $user, VoteSettings $voteSettings, Request $request)
+    {
+        //TODO: this action needs refactoring
+        $balanceVotes = $voteSettings->getVoteLimits() -
+            $this->getDoctrine()->getRepository(User::class)
+                ->getUserVotesBySettingVote($voteSettings, $user);
```

Comment:
make it in 1 line. its no way to refactor this

Reply:
I wrote about refactoring for all action